include/solvers/signed_distance_transformation.h

```diff
@@ -0,0 +1,164 @@
+   *
+   * @return Value of the computed phase fraction after applying the
+   * transformation.
+   */
+  virtual double
+  transfom_signed_distance(const double signed_distance) = 0;
+};
+
+/**
+ * @brief Transforms the signed distance to the phase fraction with a hyperbolic
+ * tangent function.
+ *
+ * The  phase fraction is defined as:
+ *
+ * \f$\phi = 0.5 - 0.5 \tanh(d/\varepsilon)\f$
+ *
+ * where \f$d\f$ is the signed distance and \f$\varepsilon\f$
+ * is a model parameter giving a measure of the interface thickness.
+ *
+ */
+class SignedDistanceTransformationTanh : public SignedDistanceTransformationBase
+{
+public:
+  /**
+   * @brief Constructor of the hyperbolic tangent signed distance transformation.
+   *
+   * @param[in] tanh_thickness Value of the \f$\varepsilon\f$ parameter
+   * modulating the interface thickness.
+   */
+  SignedDistanceTransformationTanh(const double tanh_thickness)
+    : tanh_thickness(tanh_thickness)
+  {}
+
+  /**
+   * @brief Computes the value of the phase fraction from the signed distance.
+   *
+   * @param[in] signed_distance Value of the signed distance
+   *
+   * @return Value of the computed phase fraction after applying the
+   * transformation.
+   */
+  double
+  transfom_signed_distance(const double signed_distance) override
+  {
+    return 0.5 - 0.5 * tanh(signed_distance / tanh_thickness);
+  }
+
+private:
+  /**
+   * User-defined parameter that defines the thickness of the interface. It
+   * influences its sharpness: a smaller value leads to a sharper interface.
+   */
+  const double tanh_thickness;
+};
+
+/**
+ * @brief Transforms the signed distance to the phase fraction with a piecewise
+ * polynomial function (degree 4).
+ *
+ * The phase fraction is defined as:
+ *
+ * if \f$d < 0 \f$
+ * \f$\phi = 0.5 - 0.5(4d' + 6d'^2 + 4*d'^3 + d'^4)\f$
+ *
+ * else
+ * \f$\phi = 0.5 - 0.5(4d' - 6d'^2 + 4*d'^3 - d'^4)\f$
+ *
+ * with \f$d' = d/d_\mathrm{max}\f$, where \f$d\f$ the signed distance and
+ * \f$d_\mathrm{max}\f$ the maximum redistanciation distance. This
+ * transformation clamps the phase fraction to 0 or 1 when \f$d = \pm
+ * d_\mathrm{max}\f$.
+ *
+ */
+class SignedDistanceTransformationPiecewisePolynomial
+  : public SignedDistanceTransformationBase
+{
+public:
+  /**
+   * @brief Constructor of the piecewise polynomial signed distance
+   * transformation.
+   *
+   * @param[in] max_reinitialization_distance Value of the maximum distance
+   */
+  SignedDistanceTransformationPiecewisePolynomial(
+    const double max_reinitialization_distance)
+    : max_reinitialization_distance_inv(1.0 / max_reinitialization_distance)
+  {}
+
+  /**
+   * @brief Computes the value of the phase fraction from the signed distance.
+   *
+   * @param[in] signed_distance Value of the signed distance
+   *
+   * @return Value of the computed phase fraction after applying the transtion.
+   */
+  double
+  transfom_signed_distance(const double signed_distance) override
```

Comment:
It's a small function, it is worth inlining.
```suggestion
  inline double
  transfom_signed_distance(const double signed_distance) override
```